src/app/community/community-table.component.ts

```diff
@@ -0,0 +1,190 @@
+  addDeleteObservable(res, db) {
+    if (res.docs.length > 0) {
+      const doc = res.docs[0];
+      return this.couchService.delete(db + doc._id + '?rev=' + doc._rev);
+    }
+    return of({ 'ok': true });
+  }
+
+  deleteCommunity(community) {
+    // Return a function with community on its scope to pass to delete dialog
+    const { _id: id, _rev: rev } = community;
+    return this.pipeRemovePlanetUser(this.couchService.delete('communityregistrationrequests/' + id + '?rev=' + rev), community)
+    .subscribe(([ data, userRes ]) => {
+      // It's safer to remove the item from the array based on its id than to splice based on the index
+      this.requestUpdate.emit();
+      this.editDialog.close();
+    }, (error) => this.editDialog.componentInstance.message = 'There was a problem deleting this community');
+  }
+
+  pipeRemovePlanetUser(obs: any, community) {
+    return obs.pipe(
+      switchMap(data => {
+        return forkJoin([ of(data), this.removePlanetUser(community) ]);
+      })
+    );
+  }
+
+  removePlanetUser(community) {
+    return forkJoin([
+      this.couchService.post('_users/_find', { 'selector': { '_id': 'org.couchdb.user:' + community.adminName } }),
+      this.couchService.post('shelf/_find', { 'selector': { '_id': 'org.couchdb.user:' + community.adminName } })
+    ]).pipe(switchMap(([ user, shelf ]) => {
+      return forkJoin([
+        this.addDeleteObservable(user, '_users/'),
+        this.addDeleteObservable(shelf, 'shelf/')
+      ]);
+    }));
+  }
+
+  // Gives the requesting user the 'learner' role & access to all DBs (as of April 2018)
+  unlockUser(community) {
+    return this.couchService.post('_users/_find', { 'selector': { 'requestId': community._id } })
+      .pipe(switchMap(data => {
+        const user = data.docs[0];
+        return this.couchService.put('_users/' + user._id + '?rev=' + user._rev,
+          { ...user, roles: [ 'learner' ] });
+      }));
+  }
+
+  view(planet) {
+    this.viewNationDetailDialog = this.dialog.open(DialogsViewComponent, {
+      width: '600px',
+      autoFocus: false,
+      data: {
+        allData: planet,
+        title: planet.planetType === 'nation' ? 'Nation Details' : 'Community Details'
+      }
+    });
+  }
+
+  getChildPlanet(url: string) {
+    this.dialogsListService.getListAndColumns('communityregistrationrequests',
+    { 'registrationRequest': 'accepted' }, url)
+    .pipe(takeUntil(this.onDestroy$))
+    .subscribe((planets) => {
+      const data = {
+        disableSelection: true,
+        filterPredicate: filterSpecificFields([ 'name', 'code' ]),
+        ...planets };
+      this.dialogRef = this.dialog.open(DialogsListComponent, {
+        data: data,
+        height: '500px',
+        width: '600px',
+        autoFocus: false
+      });
+    });
+  }
+
+  addHubClick(planetCode, hubName) {
+    const hub = this.hubs.find((hb: any) => hb.name === hubName);
+    hub.attached.push(planetCode);
```

Comment:
looks like its pushing community to attached of hub but not updating other hub. So its resulting to have different **attachedTo** and **children**

Reply:
Oh, the problem is that `children` shouldn't be saved to the DB, it's just for display. If the community is updated we want to show up to date data so that data is fetched every time we load the table.  `attached` is the only field that should be stored.